glue/ScavengerDelegate.cpp

```diff
@@ -0,0 +1,170 @@
+/*******************************************************************************
```

Comment:
The implementation of this file was taken from the empty glue's CLI, with functions and classes renamed.